fairlearn/metrics/experimental/_grouped_metric.py

```diff
@@ -0,0 +1,224 @@
+            for c in result.columns:
+                min_val = self.by_group[c].min()
+                result[c]['overall'] = min_val
+        else:
+            result = self.by_group.groupby(level=list(range(len(self._cf_names)))).min()
+        return result
+
+    def difference(self, method='minmax'):
+        """Read a placeholder comment."""
+        subtrahend = np.nan
+        if method == 'minmax':
+            subtrahend = self.group_min()
+        elif method == 'to_overall':
+            subtrahend = self.overall
+        else:
+            raise ValueError("Unrecognised method '{0}' in difference() call".format(method))
+
+        result = None
+        if self._cf_names is None:
+            result = pd.DataFrame(index=['overall'], columns=self.by_group.columns)
+            for c in result.columns:
+                diff = (self.by_group[c] - subtrahend[c]['overall']).abs().max()
+                result[c]['overall'] = diff
+        else:
+            # It's easiest to give in to the DataFrame columns preference
+            cf_levels = list(range(len(self._cf_names)))
+            diffs = (self.by_group.unstack(level=cf_levels) -
+                     subtrahend.unstack(level=cf_levels)).abs()
+            result = diffs.max().unstack(0)
+
+        return result
+
+    def ratio(self):
+        """Read a placeholder comment."""
+        pass
+
+    def _check_feature_length(self, feature, expected_length):
+        if len(feature) != expected_length:
+            msg = _BAD_FEATURE_LENGTH.format(len(feature), expected_length)
+            raise ValueError(msg)
+
+    def _process_functions(self, metric_functions, sample_params, params):
+        func_dict = dict()
+        if isinstance(metric_functions, list):
+            # Verify the arguments
+            sp_list = np.full(len(metric_functions), fill_value=None)
+            if sample_params is not None:
+                assert isinstance(sample_params, list)
+                for sp in sample_params:
+                    assert isinstance(sp, dict)
+                assert len(metric_functions) == len(sample_params)
+                sp_list = sample_params
+            p_list = np.full(len(metric_functions), fill_value=None)
+            if params is not None:
+                assert isinstance(params, list)
+                for p in params:
+                    assert isinstance(p, dict)
+                assert len(metric_functions) == len(params)
+                p_list = params
+
+            # Iterate
+            for i in range(len(metric_functions)):
+                fc = FunctionContainer(metric_functions[i], None, sp_list[i], p_list[i])
+                assert fc.name_ not in func_dict
+                func_dict[fc.name_] = fc
+        else:
+            fc = FunctionContainer(metric_functions, None, sample_params, params)
+            func_dict[fc.name_] = fc
+        return func_dict
+
+    def _process_features(self, base_name, features, expected_length):
```

Comment:
You probably can use sklearn's `check_consistent_length` around here a bit.

Reply:
Thanks - I'll look into it.

Reply:
Have switched over, thanks!